notebook/notebookapp.py

```diff
@@ -1671,7 +1671,7 @@ def server_info(self):
     def write_server_info_file(self):
         """Write the result of server_info() to the JSON file info_file."""
         try:
-            with open(self.info_file, 'w') as f:
+            with io.open(self.info_file, 'w', encoding='utf-8') as f:
```

Comment:
Sorry, I think I told you to change the wrong one. This one needs to stay as regular `open()`, because the JSON module on Python 2 produces bytes, not unicode.

I thought we were looking at this line, but it already uses `io.open()`:

https://github.com/jupyter/notebook/blob/0579bc09699e795a8900ddd02105848f5fcb7fcc/notebook/notebookapp.py#L1699

Reply:
If commit f8d7d40 has all that is needed, can you use only that commit
and leave out the further two commits in this branch when you merge?

Reply:
Yup. I'll open a new PR so it re-runs the CI tests with just that.

Reply:
See #4349